R/SpectraWrapper.R

```diff
@@ -1,7 +1,168 @@
-#' @title SpectraWrapper
+#' @title RunSpectra
 #' @description Wrapper function for the Spectra package
 #' @param seuratObj Input Seurat object
+#' @param seuratToAdataDir Directory to house files used in generating Ann Data
+#' @param assayName Assay used for generating Ann Data
+#' @param inputJSONfile Input Seurat object
+#' @param immportfile Input Seurat object
 
-SpectraWrapper <- function(seuratObj) {
-  print("Future site of Spectra Wrapper!")
+RunSpectra <- function(seuratObj, seuratToAdataDir, assayName, inputJSONfile, immportfile){
+
+  SeuratToAData(seuratObj, seuratToAdataDir, assayName)
+  gex_datafile = paste0(seuratToAdataDir, "/GEX.h5")
+  metadata_file = paste0(seuratToAdataDir, "/metadata.tsv")
+  genesets_json = paste0(seuratToAdataDir, "/genesets.json")
+  variable_features_file = paste0(seuratToAdataDir, "varfeats.csv")
+
+  ProduceGenesetJSONfile(inputJSONfile, immportfile, genesets_json)
```

Comment:
like we talked about - the user should probably just have a single "genesets.json" file that they interact with

Reply:
afterwords, we should check that the genesets.json file is spectra compatible. 

iirc, there's a hierarchy necessary for the gene sets - something like CellType: geneset: genes